src/js/tracing/timetodisplay.tsx

```diff
@@ -0,0 +1,238 @@
+    op: 'ui.load.full_display',
+    name: 'Time To Full Display',
+    startTimestamp: spanToJSON(initialDisplaySpan).start_timestamp,
+    ...options,
+  });
+  if (!fullDisplaySpan) {
+    return;
+  }
+
+  const timeout = setTimeout(() => {
+    if (spanToJSON(fullDisplaySpan).timestamp) {
+      return;
+    }
+    fullDisplaySpan.setStatus('deadline_exceeded');
+    fullDisplaySpan.end(spanToJSON(initialDisplaySpan).timestamp);
+    setSpanDurationAsMeasurement('time_to_full_display', fullDisplaySpan);
+    logger.warn(`[TimeToDisplay] Full display span deadline_exceeded.`);
+  }, options.timeoutMs);
+
+  fill(fullDisplaySpan, 'end', (originalEnd: SpanClass['end']) => (endTimestamp?: Parameters<SpanClass['end']>[0]) => {
+    clearTimeout(timeout);
+    originalEnd.call(fullDisplaySpan, endTimestamp);
+  });
+
+  return fullDisplaySpan;
+}
+
+function onDrawNextFrame(event: { nativeEvent: RNSentryOnDrawNextFrameEvent }): void {
+  logger.debug(`[TimeToDisplay] onDrawNextFrame: ${JSON.stringify(event.nativeEvent)}`);
+  if (event.nativeEvent.type === 'fullDisplay') {
+    return updateFullDisplaySpan(event.nativeEvent.newFrameTimestampInSeconds);
+  }
+  if (event.nativeEvent.type === 'initialDisplay') {
+    return updateInitialDisplaySpan(event.nativeEvent.newFrameTimestampInSeconds);
+  }
+}
+
+function updateInitialDisplaySpan(frameTimestampSeconds: number): void {
+  const span = startTimeToInitialDisplaySpan();
+  if (!span) {
+    logger.warn(`[TimeToDisplay] No span found or created, possibly performance is disabled.`);
+    return;
+  }
+
+  const activeSpan = getActiveSpan();
+  if (!activeSpan) {
+    logger.warn(`[TimeToDisplay] No active span found to attach ui.load.initial_display to.`);
+    return;
+  }
+
+  if (spanToJSON(span).parent_span_id !== spanToJSON(activeSpan).span_id) {
+    logger.warn(`[TimeToDisplay] Initial display span is not a child of current active span.`);
+    return;
+  }
+
+  if (spanToJSON(span).timestamp) {
+    logger.warn(`[TimeToDisplay] ${spanToJSON(span).description} span already ended.`);
+    return;
+  }
+
+  span.end(frameTimestampSeconds);
+  span.setStatus('ok');
+  logger.debug(`[TimeToDisplay] ${spanToJSON(span).description} span updated with end timestamp.`);
+
+  setSpanDurationAsMeasurement('time_to_initial_display', span);
+}
+
+function updateFullDisplaySpan(frameTimestampSeconds: number): void {
+  const span = startTimeToFullDisplaySpan();
+  if (!span) {
+    logger.warn(`[TimeToDisplay] No span found or created, possibly performance is disabled.`);
+    return;
+  }
+
+  if (spanToJSON(span).timestamp) {
+    logger.warn(`[TimeToDisplay] ${spanToJSON(span).description} span already ended.`);
+    return;
+  }
+
+  span.end(frameTimestampSeconds);
+
+  span.setStatus('ok');
+  logger.debug(`[TimeToDisplay] ${spanToJSON(span).description} span updated with end timestamp.`);
+
+  setSpanDurationAsMeasurement('time_to_full_display', span);
+}
```

Comment:
h: We need to check whether TTID span is finished already, otherwise we need to wait for it and then automatically close TTFD with the same end.

Reply:
Thank you, I fixed it here https://github.com/getsentry/sentry-react-native/pull/3654/commits/c06e638609e43b8bf1b0263fbea25dfc26a251ad